From 829ca990e84d441817c77d89e223d442f2d26eae Mon Sep 17 00:00:00 2001
From: Jeff Moyer <jmoyer@redhat.com>
Date: Thu, 28 Aug 2014 19:16:05 -0400
Subject: [virt] virtio_blk, blk-mq: add async parameter to blk_mq_start_stopped_hw_queues

Message-id: <1409253371-23161-16-git-send-email-jmoyer@redhat.com>
Patchwork-id: 89864
O-Subject: [RHEL7 PATCH 15/21] virtio-blk, blk-mq: add async parameter to blk_mq_start_stopped_hw_queues
Bugzilla: 1105698
RH-Acked-by: Fam Zheng <famz@redhat.com>

This is a backport of the virtio-blk portion of the following commit.
This is related to bug 1105698.

  commit 1b4a325858f695a9b5041313602d34b36f463724
  Author: Christoph Hellwig <hch@lst.de>
  Date:   Wed Apr 16 09:44:54 2014 +0200

    blk-mq: add async parameter to blk_mq_start_stopped_hw_queues

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/virtio_blk.c b/drivers/block/virtio_blk.c
index f909a88..7a51f06 100644
--- a/drivers/block/virtio_blk.c
+++ b/drivers/block/virtio_blk.c
@@ -151,7 +151,7 @@ static void virtblk_done(struct virtqueue *vq)
 
  /* In case queue is stopped waiting for more buffers. */
  if (req_done)
-  blk_mq_start_stopped_hw_queues(vblk->disk->queue);
+  blk_mq_start_stopped_hw_queues(vblk->disk->queue, true);
 }
 
 static int virtio_queue_rq(struct blk_mq_hw_ctx *hctx, struct request *req)
@@ -762,7 +762,7 @@ static int virtblk_restore(struct virtio_device *vdev)
  vblk->config_enable = true;
  ret = init_vq(vdev->priv);
  if (!ret)
-  blk_mq_start_stopped_hw_queues(vblk->disk->queue);
+  blk_mq_start_stopped_hw_queues(vblk->disk->queue, true);
 
  return ret;
 }
-- 
1.7.1